### PR TITLE
[RSDK-7992] Add keepalive

### PR DESCRIPTION
**Pull request by npmenard**
Add a keep alive timeout for both the HTTP2 layer and the Socket layer

### CODE REVIEW

File: micro-rdk/src/esp32/tls.rs [hunk 1]
```diff
@@ -261,6 +264,56 @@ impl Esp32TLSStream {
                             let mut fd: i32 = 0;
                             esp_idf_svc::sys::esp!(esp_tls_get_conn_sockfd(*tls_context, &mut fd))
                                 .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
+
+                            // set socket keep alive properties as such
+                            // KEEPIDLE is set to 120 second (time before a first keepalive probe is sent)
+                            // KEEPINTVL, KEEPCNT are set to 60 and 4 respectively
+                            // Total time before an IDLE and DEAD connection is closed =  360s
+                            let var: i32 = 1;
+                            if lwip_setsockopt(
```
Comment (mattjperez) [inline micro-rdk/src/esp32/tls.rs:273]: what does this first option set? enables keepalive?

File: micro-rdk/src/esp32/tls.rs [hunk 2]
```diff
@@ -261,6 +264,56 @@ impl Esp32TLSStream {
...
+                            {
+                                return Err(std::io::Error::last_os_error());
+                            }
+                            let var: i32 = 60;
+                            if lwip_setsockopt(
+                                fd,
+                                IPPROTO_TCP as i32,
+                                TCP_KEEPINTVL as i32,
+                                &var as *const i32 as *const _,
+                                std::mem::size_of::<i32>() as socklen_t,
```
Comment (mattjperez) [inline micro-rdk/src/esp32/tls.rs:289]: Is it possible to declare these values as `const` somewhere (ie `const KEEPALIVE_INTVL_VAL`) and reference them directly (ie on L288 `&KEEPALIVE_INVL_VAL as *const i32 as *const _,...`)

**Review by mattjperez — APPROVED** [review]
some non-blocking questions, but LGTm

**Review by acmorrow — APPROVED** [review]
LGTM

File: micro-rdk/src/esp32/tls.rs (hunk 1 above)
Comment (npmenard) [reply micro-rdk/src/esp32/tls.rs:277]: yup

File: micro-rdk/src/esp32/tls.rs (hunk 2 above)
Comment (npmenard) [reply micro-rdk/src/esp32/tls.rs]: make sense

**Pull request merged**